geode-core/src/integrationTest/java/org/apache/geode/pdx/JSONFormatterJUnitTest.java

```diff
@@ -210,4 +231,214 @@ public void testJSONStringSortedFields() {
       System.setProperty(JSONFormatter.SORT_JSON_FIELD_NAMES_PROPERTY, "false");
     }
   }
+
+  @Test
+  public void testReorderJSONFieldsDoesNotCauseManyTypeIdConflicts() {
```

Comment:
Possibly break this down into smaller sets of tests.  Prove when using the system property for sorted JSON field objects, we only produce 1 type

When there are collisions to the same calculated id, we do not lose a type